downloader.go

```diff
@@ -21,61 +23,58 @@ type Downloader struct {
 	cacheDir string
 }
 
+var schemeRegexp = regexp.MustCompile(`^.*://.*`)
+
 func NewDownloader(logger logging.Logger, cacheDir string) *Downloader {
 	return &Downloader{
 		logger:   logger,
 		cacheDir: cacheDir,
 	}
 }
 
-func (d *Downloader) Download(uri string) (string, error) {
-	url, err := url.Parse(uri)
-	if err != nil {
-		return "", err
-	}
+func (d *Downloader) Download(pathOrUri string) (string, error) {
 
-	switch url.Scheme {
-	case "", "file":
-		return d.handleFile(url)
-	case "http", "https":
-		return d.handleHTTP(uri)
-	default:
-		return "", fmt.Errorf("unsupported protocol in URI %q", uri)
-	}
-}
-
-func (d *Downloader) handleFile(bpURL *url.URL) (string, error) {
-	path := bpURL.Path
-
-	if filepath.Ext(path) != ".tgz" {
-		return path, nil
-	}
+	hasScheme := schemeRegexp.MatchString(pathOrUri)
```

Comment:
I'm curious why we broke the no scheme case out of the switch statement. Was there some edge case here I am not comprehending?

Reply:
This is because when we rely on `url.Parse(pathOrUri)` the result for `c:\user\test\` _incorrectly_ picks up `c` as the scheme later falling into the `unsupported protocol`. This new check is more for determining if it's a uri or file path so we can be certain `url.Parse` will yield more accurate results. 

Would it help to rename the boolean to `isUri`?